openam-authentication/openam-auth-fr-oath/src/main/java/org/forgerock/openam/authentication/modules/fr/oath/AuthenticatorOATH.java

```diff
@@ -465,10 +469,12 @@ private OathDeviceSettings createBasicDevice() throws AuthLoginException {
         OathDeviceSettings settings = oathDevices.createDeviceProfile(minSecretKeyLength);
         settings.setChecksumDigit(checksum);
 
-        try {
-            settings.setRecoveryCodes(recoveryCodeGenerator.generateCodes(NUM_CODES, Alphabet.ALPHANUMERIC, false));
-        } catch (CodeException e) {
-            throw new AuthLoginException(amAuthOATH, "authFailed", null);
+        if (this.issueRecoveryCodesEnabled) {
```

Comment:
I prefer `issueRecoveryCodesEnabled` than `this.issueRecoveryCodesEnabled` here.

Reply:
I think the same way.

There was no problem except this indication.
Can I wait for a fix, or fix it here and merge it?

Reply:
Sorry for the wait. Please merge this.